src/pages/event.js

```diff
@@ -70,59 +68,36 @@ export default function Events() {
 export function Event() {
   const params = useParams();
   const { eventId } = params;
-  const dispatch = useDispatch();
   const { trackPageView, trackLink } = useMatomo();
 
-  const tokens = useSelector((state) => state.events.tokens);
-  const loadingEvent = useSelector((state) => state.events.eventStatus);
-  const errorEvent = useSelector((state) => state.events.eventError);
-  const event = useSelector((state) => state.events.event);
-
   const [pageIndex, setPageIndex] = useState(0);
+  const [event, setEvent] = useState(undefined);
+  const [tokens, setTokens] = useState([]);
+  const [loadingTokensFailed, setLoadingTokensFailed] = useState(false);
   const [csv_data, setCsv_data] = useState([]);
-  const [ensNames, setEnsNames] = useState([]);
   const [canDownloadCsv, setCanDownloadCsv] = useState(CSV_STATUS.NoTokens);
   const [tableIsLoading, setTableIsLoading] = useState(true);
   const [trackedEvent, setTrackedEvent] = useState(null);
-  const pageCount = useMemo(
-    () =>
-      event.tokenCount % 50 !== 0
-        ? Math.floor(event.tokenCount / 50) + 1
-        : event.tokenCount,
-    [event]
-  );
   const power = calculatePower(csv_data);
 
   const csvDownloading = () =>
     canDownloadCsv === CSV_STATUS.DownloadingLastDataChunk ||
     canDownloadCsv === CSV_STATUS.DownloadingData;
   const csvReady = () => canDownloadCsv === CSV_STATUS.Ready;
-  const csvOnlyMissingEns = () => canDownloadCsv === CSV_STATUS.ReadyWithoutEns;
   const csvFailed = () => canDownloadCsv === CSV_STATUS.Failed;
 
-  const readyToResolveENS = () =>
-    canDownloadCsv === CSV_STATUS.DownloadingLastDataChunk ||
-    canDownloadCsv === CSV_STATUS.ReadyWithoutEns ||
-    canDownloadCsv === CSV_STATUS.Ready ||
-    canDownloadCsv === CSV_STATUS.Failed;
-
-  const succeededLoadingEvent = () =>
-    loadingEvent === FETCH_EVENT_PAGE_INFO_STATUS.SUCCEEDED;
-  const isLoadingEvent = () =>
-    loadingEvent === FETCH_EVENT_PAGE_INFO_STATUS.LOADING;
-  const failedLoadingEvent = () =>
-    loadingEvent === FETCH_EVENT_PAGE_INFO_STATUS.FAILED;
-  const isIdle = () =>
-    loadingEvent === FETCH_EVENT_PAGE_INFO_STATUS.IDLE ||
-    loadingEvent === undefined;
-
   useEffect(() => {
     window.scrollTo(0, 0);
   }, []);
 
+  useEffect(() => {
+    if (event) return;
+
+    getDrop(eventId).then(setEvent);
+  }, [event]);
```

Comment:
Shouldn't `eventId` and `setEvent` also be in dependencies?

Reply:
`setEvent` is the function that modify the state so it's stable it will not change so it doesn't make sense to include it